### PR TITLE
Replace string cats with join() in pgfplots (and more).

### PR DESCRIPTION
**Pull request by pkofod**
Besides in a few places.

### CODE REVIEW

File: src/backends/pgfplots.jl [hunk 1]
```diff
@@ -175,52 +175,55 @@ end
...
     _pgfplots_get_label!(kwargs, plt)
+    kwargs[:style] = join(kwargs[:style], ',')
     kwargs
 end
 
+
+
 function _pgfplots_axis(plt_series)
     line_type = plt_series[:linetype]
     plt_kwargs = _pgfplots_get_plot_kwargs(plt_series)
```
Comment (pkofod) [inline src/backends/pgfplots.jl]: this line returns a `plt_kwargs[:style]` which has been joined... (continued below)

File: src/backends/pgfplots.jl [hunk 2]
```diff
@@ -175,52 +175,55 @@ end
...
-        if line_type == :path
+    if line_type == :path
         PGFPlots.Linear(plt_series[:x], plt_series[:y]; plt_kwargs...)
     elseif line_type == :path3d
         PGFPlots.Linear3(plt_series[:x], plt_series[:y], plt_series[:z]; plt_kwargs...)
     elseif line_type == :scatter
         PGFPlots.Scatter(plt_series[:x], plt_series[:y]; plt_kwargs...)
     elseif line_type == :steppre
-        plt_kwargs[:style] *= "const plot mark right,"
+        plt_kwargs[:style] *= ", const plot mark right"
```
Comment (pkofod) [inline src/backends/pgfplots.jl]: ... (continued from above) so this line has to use `*=`. If not, I would have to do the "join" in all the `elseif` blocks.

**Comment by pkofod** [comment]
Please let me know what you think / if something can be done better.

File: src/backends/pgfplots.jl [hunk 3]
```diff
@@ -87,17 +87,17 @@ const _pgfplots_linestyles = KW(
...
+    :cross => "mark = +",
+    :xcross => "mark = x",
+    :utriangle => "mark = triangle*",
+    :dtriangle => "mark = triangle*",
+    :ellipse => "mark = o*",
+    :rect => "mark = square*",
+    :star5 => "mark = star",
+    :star6 => "mark = asterisk",
+    :diamond => "mark = diamond*",
+    :pentagon => "mark = pentagon*"
```
Comment (tbreloff) [inline src/backends/pgfplots.jl]: I'll just make some notes of what I might do if I was cleaning up (you don't have to listen to me though).

Here it seems wasteful to include "mark = " in every string when you could do that once outside of the KW.

File: src/backends/pgfplots.jl [hunk 4]
```diff
@@ -120,50 +120,50 @@ end
...
     # Control marker colors and alphas
     α = plt[:markeralpha] == nothing ? 1.0 : plt[:markeralpha]
-    kwargs[:style] *= "mark options = {color=$(_pgfplots_get_color(plt, :markerstrokecolor)),"
-    kwargs[:style] *= mark == :dtriangle ? "rotate=180," : ""
-    kwargs[:style] *= "fill=$(_pgfplots_get_color(plt, :markercolor)),"
-    kwargs[:style] *= "fill opacity = $α,"
+    push!(kwargs[:style], "mark options = {color=$(_pgfplots_get_color(plt, :markerstrokecolor))")
+    mark == :dtriangle && push!(kwargs[:style], "rotate=180")
+    push!(kwargs[:style], "fill=$(_pgfplots_get_color(plt, :markercolor))")
+    push!(kwargs[:style], "fill opacity = $α")
```
Comment (tbreloff) [inline src/backends/pgfplots.jl]: Could you combine some of these together?  `"fill = ..., fill opacity = ..."`

On that note, will PGFPlots accept a multiline style?

**Comment by tbreloff** [comment]
I'm fine with whatever you decide to do here, as long as it still works! :+1:

File: src/backends/pgfplots.jl (hunk 3 above)
Comment (pkofod) [inline src/backends/pgfplots.jl]: :+1:

File: src/backends/pgfplots.jl (hunk 4 above)
Comment (pkofod) [inline src/backends/pgfplots.jl]: I certainly could, I was just trying to keep the line width sane. What is a multiline?

**Comment by tbreloff** [comment]
The multiline comment was about doing tsomething like:

```julia
using PGFPlots
Plots.Linear(1:10, sin(1:10), style="""
ybar,
fill = green,
mark = +,
,
mark size=10
""")
```

And so you could do blocks like:

```julia
push!(kwargs["style], """
  mark = $(...),
  fill = $(...),
  $(haskey(...) ? "something = something" : ""),
  ...
""")
```

**Comment by tbreloff** [comment]
It's just the wall of `push!(kwargs[:style], ...)` is hard to look at.

**Comment by pkofod** [comment]
> The multiline comment was about doing...
When you said "multiline style" I thought you mean a line style... My bad.

**Comment by tbreloff** [comment]
I'll take your lead here... let me know when you want to merge.  (maybe do a `git fetch` and `git rebase origin/dev` first?)

**Comment by pkofod** [comment]
Ill rebase when I'm at a computer.

**Comment by pkofod** [comment]
It's rebased.

**Comment by tbreloff** [comment]
Cool thanks

**Pull request merged**